test/scripts/check-siblingsHelper.ts

```diff
@@ -0,0 +1,221 @@
+  const separator = process.platform === 'win32' ? ';' : ':';
+  const pathArray = path.split(separator);
+  const newPath = pathArray
+    .filter((item) => !item.startsWith('/tmp/yarn--'))
+    .join(separator);
+  return newPath;
+}
+
+
+export function GetPackageManagerVersion(pkgMnger: string, testPath: string): string {
+  // Check if corepack is working/
+  const newEnv = {
+    ...process.env,
+    PATH: HideGlobalYarnPath(),
+    INIT_CWD: testPath
+  };
+
+  const result2 = spawnSync('corepack', ['--version'], {
+    cwd: testPath,
+    stdio: ['pipe'], // Ensure output is in readable string format
+    // Clear env to avoid contamination with root folder.
+    env: {
+      ...newEnv
+    }
+  });
+
+  if (result2.error) {
+    throw new Error(` Corepack Failed.
+      SpawnSync Error: ${result2.error}
+      Result object: ${JSON.stringify(result2, null, 2)}
+
+      STDERR: ${result2.stderr?.toString()}
+
+      STDOUT: ${result2.stdout?.toString()}`);
+  }
+  expect(result2.status).toBe(0);
+  expect(result2.stderr?.toString()).toBeEmpty();
+  expect(result2.stdout.toString().trim()).not.toBeEmpty();
+
+  // Check if it is running the correct package manager.
+  const result = spawnSync(pkgMnger, ['--version'], {
+    cwd: testPath,
+    stdio: ['pipe'], // Ensure output is in readable string format
+    // Clear env to avoid contamination with root folder.
+    env: {
+      ...newEnv
+    }
+  });
+
+  if (result.error) {
+    throw new Error(` Corepack Failed.
+      SpawnSync Error: ${result.error}
+      Result object: ${JSON.stringify(result, null, 2)}
+
+      STDERR: ${result.stderr?.toString()}
+
+      STDOUT: ${result.stdout?.toString()}`);
+  }
+  expect(result.status).toBe(0);
+  expect(result.stderr?.toString()).toBeEmpty();
+
+  return result.stdout.toString().trim();
+}
+
+function filterYarnV3Log(str: string): string {
+  const regex = /(?<=➤ YN\d{4}: [│┌└] ).*/;
+
+  const filterV3E2ELogs = (str: string) => {
+    const v3Regex = /2e2-sibling-test@workspace:. (STDOUT|STDERR) /g;
+    return str.replace(v3Regex, '');
+  };
+
+  const filterNPMLE2ELogs = (str: string) => {
+    const npmRegex = /npm error /g;
+    return str.replace(npmRegex, '');
+  };
+
+  const match = str.match(regex);
+  // ➤ YN0000: │ @sentry/capacitor...
+  // to
+  // @sentry/capacitor...
+  return match ? filterV3E2ELogs(match[0]) : filterNPMLE2ELogs(str);
+}
+
+// For testing.
+export function FilterLogs(data: string, filter: boolean): string[] {
+  return data.split('\n').map(filterYarnV3Log).filter(log => filter == log.startsWith('E2E_TEST'));
+}
+
+export function GetE2ELogs(testPath: string): string[] {
+  const data = fs.readFileSync(path.join(testPath, installLogFilename), 'utf8');
+  return FilterLogs(data, true);
+}
+
+export function GetLogs(testPath: string): string[] {
+  const data = fs.readFileSync(path.join(testPath, installLogFilename), 'utf8');
+  return FilterLogs(data, false);
+}
```

Comment:
Do we need the filtering logic? If I understand it correctly we anyway use `contains` to check if the message is present in the logs, so I would expect it to pass also without the filtering.